internal/core/item_tree.rs

```diff
@@ -95,6 +95,26 @@ impl ItemRc {
         unsafe { core::mem::transmute::<Pin<ItemRef<'_>>, Pin<ItemRef<'a>>>(result) }
     }
 
+    /// Returns a `VRcMapped` of this item, to conveniently access specialized item API.
+    pub fn downcast<'a, T: HasStaticVTable<ItemVTable>>(
+        &'a self,
+    ) -> Option<VRcMapped<ComponentVTable, T>> {
+        #![allow(unsafe_code)]
+        let item = self.borrow();
+        if ItemRef::downcast_pin::<T>(item).is_none() {
+            return None;
+        }
```

Comment:
Why this short circuit.  The map_dyn should anyway be cheap

Reply:
It's not about the performance, it's about the "safety", i.e. what if the item can't be downcasted to the desired type?